lib/CTownHandler.cpp

```diff
@@ -894,6 +894,11 @@ void CTownHandler::loadTown(CTown * town, const JsonNode & source)
 		town->namesCount += 1;
 	}
 
+	VLC->modh->identifiers.requestIdentifier(source["moatAbility"], [=](si32 ability)
+	{
+		town->moatAbility = SpellID(ability);
+	});
+

```

Comment:
Preferred form here is
`VLC->modh->identifiers.requestIdentifier( "spell", source["moatAbility"], [=](si32 ability)`
Code only accepts spells, so you can specifically request for spell identifier

Reply:
I think we don't?
`modh` however is very different from other handlers/services, so might need different access model than services

Reply:
GeneralText is (luckily) quite easy one - all we need to do is to provide "translator" service that will call translate() method of GeneralText.

Mod Handler however is much more massive and way more integrated into engine.
Perhaps have it provide multiple "services"? E.g. "identifiers" service, mod information services, and maybe others. At least for now can be done without massive refactoring - meaning, make CModHandler implement multiple interfaces for all provided services.

That being said, extracting identifiers handling from ModHandler might not be a bad idea.